evmcore/tx_validation_test.go

```diff
@@ -0,0 +1,790 @@
+			require.NoError(t, err)
+			setGas(t, tx, floorDataGas-1)
+			opt.currentMaxGas = floorDataGas
+
+			// --- needed for execution up to relevant check ---
+			// set tx for execution
+			setGasPriceOrFeeCap(t, tx, opt.minTip)
+			// sign txs with sender
+			address, signedTx := signTxForTest(t, tx)
+			// setup enough balance
+			testDb := newTestTxPoolStateDb()
+			testDb.balances[address] = uint256.NewInt(math.MaxUint64)
+			opt.currentState = testDb
+			// ---
+
+			// validate transaction
+			err = validateTx(signedTx, opt)
+			require.NoError(t, err)
+
+		})
+	}
+}
+
+func TestValidateTx_Data_RejectsTxWith(t *testing.T) {
+	oversizedData := make([]byte, txMaxSize+1)
+	for name, tx := range getTxsOfAllTypes() {
+		t.Run(fmt.Sprintf("oversized data/%v", name), func(t *testing.T) {
+
+			setData(t, types.TxData(tx), oversizedData)
+
+			err := validateTx(types.NewTx(tx), getTestTransactionsOption())
+			require.Equal(t, ErrOversizedData, err)
+		})
+	}
+
+	// EIP-3860
+	maxInitCode := make([]byte, params.MaxInitCodeSize+1)
+	for name, tx := range getTxsOfAllTypes() {
+		t.Run(fmt.Sprintf("init code too large/%v", name), func(t *testing.T) {
+			if isBlobOrSetCode(tx) {
+				t.Skip("blob and setCode transactions cannot be used as create")
+			}
+
+			setData(t, types.TxData(tx), maxInitCode)
+			setReceiverToNil(t, tx)
+
+			err := validateTx(types.NewTx(tx), getTestTransactionsOption())
+			require.ErrorIs(t, err, ErrMaxInitCodeSizeExceeded)
+		})
+	}
+
+	for name, tx := range getTxsOfAllTypes() {
+		t.Run(fmt.Sprintf("init code size not checked before shanghai/%v", name), func(t *testing.T) {
+			if isBlobOrSetCode(tx) {
+				t.Skip("blob and setCode transactions cannot be used before eip4844 and eip7702")
+			}
+			opt := getTestTransactionsOption()
+			opt.shanghai = false
+			opt.eip4844 = false
+			opt.eip7623 = false
+			// needs extra gas to allow big data to be afforded.
+			opt.currentMaxGas = 249_612
+
+			setData(t, types.TxData(tx), maxInitCode)
+			setReceiverToNil(t, tx)
+
+			setGasPriceOrFeeCap(t, tx, opt.currentBaseFee)
+			setGas(t, tx, opt.currentMaxGas) // enough gas
+			address, signedTx := signTxForTest(t, tx)
+			testDb := newTestTxPoolStateDb()
+			testDb.balances[address] = uint256.NewInt(opt.currentMaxGas*opt.currentBaseFee.Uint64() + 1)
+			opt.currentState = testDb
+
+			err := validateTx(signedTx, opt)
+			require.NoError(t, err)
+		})
+	}
+
+}
+
+func TestValidateTx_Signer_RejectsTxWith(t *testing.T) {
+	for name, tx := range getTxsOfAllTypes() {
+		t.Run(fmt.Sprintf("invalid signature/%v", name), func(t *testing.T) {
+			opt := getTestTransactionsOption()
+			opt.signer = types.HomesteadSigner{}
+			setSignatureValues(t, tx, big.NewInt(1), big.NewInt(2), big.NewInt(3))
+			err := validateTx(types.NewTx(tx), getTestTransactionsOption())
+			require.ErrorIs(t, err, ErrInvalidSender)
+		})
+	}
+
+	for name, tx := range getTxsOfAllTypes() {
+		t.Run(fmt.Sprintf("invalid signer/%v", name), func(t *testing.T) {
```

Comment:
please check: I think this test actually checks for an invalid chain-Id.
```suggestion
		t.Run(fmt.Sprintf("invalid chain-id/%v", name), func(t *testing.T) {
```